common/Resources/ti.internal/extensions/node/internal/util/types.js

```diff
@@ -0,0 +1,237 @@
+
+export function isAsyncFunction(value) {
+	return checkPrototype(value, 'AsyncFunction');
+}
+
+export function isBigInt64Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'BigInt64Array';
+}
+
+export function isBigUint64Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'BigUint64Array';
+}
+
+export function isBooleanObject(value) {
+	return checkPrototype(value, 'Boolean');
+}
+
+export function isBoxedPrimitive(value) {
+	if (typeof value !== 'object') {
+		return false;
+	}
+
+	return isNumberObject(value)
+		|| isStringObject(value)
+		|| isBooleanObject(value)
+		// || isBigIntObject(value)
+		|| isSymbolObject(value);
+}
+
+export function isDataView(value) {
+	return checkPrototype(value, 'DataView');
+}
+
+export function isDate(value) {
+	return checkPrototype(value, 'Date');
+}
+
+// @todo isExternal
+
+export function isFloat32Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'Float32Array';
+}
+
+export function isFloat64Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'Float64Array';
+}
+
+export function isGeneratorFunction(value) {
+	return checkPrototype(value, 'GeneratorFunction');
+}
+
+export function isGeneratorObject(value) {
+	return checkPrototype(value, 'GeneratorObject');
+}
+
+export function isInt8Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'Int8Array';
+}
+
+export function isInt16Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'Int16Array';
+}
+
+export function isInt32Array(value) {
+	return TypedArrayProto_toStringTag(value) === 'Int32Array';
+}
+
+export function isMap(value) {
+	return checkPrototype(value, 'Map');
+}
+
+export function isMapIterator(value) {
+	if (typeof value !== 'object') {
+		return false;
+	}
+
+	const prototype = Object.getPrototypeOf(value);
+	return prototype && prototype[Symbol.toStringTag] === 'Map Iterator';
+}
+
+// @todo isModuleNamespaceObject
+
+export function isNativeError(value) {
+	// if not an instance of an Error, definitely not a native error
+	if (!(value instanceof Error)) {
+		return false;
+	}
+	if (!value || !value.constructor) {
+		return false;
+	}
+	return [
+		'Error',
+		'EvalError',
+		'RangeError',
+		'ReferenceError',
+		'SyntaxError',
+		'TypeError',
+		'URIError'
+	].includes(value.constructor.name);
+}
+
+export function isNumberObject(value) {
+	return checkPrototype(value, 'Number');
+}
+
+export function isPromise(value) {
+	return checkPrototype(value, 'Promise');
+}
+
+// @todo isProxy
+
+export function isRegExp(value) {
+	return checkPrototype(value, 'RegExp');
+}
+
+export function isSet(value) {
+	return checkPrototype(value, 'Set');
+}
+
+export function isSetIterator(value) {
+	if (typeof value !== 'object') {
+		return false;
+	}
+
+	const prototype = Object.getPrototypeOf(value);
+	return prototype && prototype[Symbol.toStringTag] === 'Set Iterator';
+}
+
+export function isSharedArrayBuffer(value) {
+	if (!global.SharedArrayBuffer) {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/internal/util/types.js;
  Line: 189;
-->

- :warning: common/Resources/ti.internal/extensions/node/internal/util/types.js line 189 – The 'SharedArrayBuffer' is not supported until Node.js 8.10.0. The configured version range is '>=8'. (node/no-unsupported-features/es-builtins)